integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py

```diff
@@ -0,0 +1,153 @@
+test_dataset = test_dataset.batch(params["batch_size"])
+
+# Prepare model
+# Model
+model = tf.keras.models.Sequential(
+    [
+        tf.keras.layers.Flatten(input_shape=(28, 28)),
+        tf.keras.layers.Dense(128, activation="relu"),
+        tf.keras.layers.Dense(10),
+    ]
+)
+
+# Loss
+loss_object = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+
+# Optimizer
+optimizer = tf.keras.optimizers.Adam(params["lr"])
+
+
+# (Neptune) Log model summary
+with io.StringIO() as s:
+    model.summary(print_fn=lambda x: s.write(x + "\n"))
+    model_summary = s.getvalue()
+
+run["training/model/summary"] = model_summary
+
+
+# Helper functions for training loop
+def loss_and_preds(model, x, y, training):
+    # training=training is needed only if there are layers with different
+    # behavior during training versus inference (e.g. Dropout).
+    y_ = model(x, training=training)
+
+    return loss_object(y_true=y, y_pred=y_), y_
+
+
+def grad(model, inputs, targets):
+    with tf.GradientTape() as tape:
+        loss_value, _ = loss_and_preds(model, inputs, targets, training=True)
+    return loss_value, tape.gradient(loss_value, model.trainable_variables)
+
+
+# Training Loop
+for epoch in range(params["num_epochs"]):
+    epoch_loss_avg = tf.keras.metrics.Mean()
+    epoch_accuracy = tf.keras.metrics.SparseCategoricalAccuracy()
+
+    for x, y in train_dataset:
+        loss_value, grads = grad(model, x, y)
+        optimizer.apply_gradients(zip(grads, model.trainable_variables))
+
+        epoch_loss_avg.update_state(loss_value)
+        epoch_accuracy.update_state(y, model(x, training=True))
+
+    # (Neptune) Log metrics for the epoch
+    # Train metrics
+    run["training/train/loss"].log(epoch_loss_avg.result())
+    run["training/train/accuracy"].log(epoch_accuracy.result())
+
+    # (Neptune) Log test metrics
+    test_loss, test_preds = loss_and_preds(model, test_examples, test_labels, False)
+    run["training/test/loss"].log(test_loss)
+    test_acc = epoch_accuracy(test_labels, test_preds)
+    run["training/test/accuracy"].log(test_acc)
+
+    # (Neptune) Log test prediction
+    for idx in range(params["num_visualization_examples"]):
+        np_image = test_examples[idx].numpy().reshape(28, 28)
+        image = neptune.types.File.as_image(np_image)
+        pred_label = test_preds[idx].numpy().argmax()
+        true_label = test_labels[idx]
+        run[f"training/visualization/epoch_{epoch}"].log(
+            image, description=f"pred={pred_label} | actual={true_label}"
+        )
+
+    if epoch % 5 == 0 or epoch == (params["num_epochs"] - 1):
+        print(
+            "Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(
+                epoch, epoch_loss_avg.result(), epoch_accuracy.result()
+            )
+        )
+
+# Tracking model with Neptune model registry
+# Refer to the [documentation](https://neptune.ai/product/model-registry) for more information
```

Comment:
Could have been me that suggested this should be Markdown link formatted -- sorry about that. Realized now this is a script, not notebook.

```suggestion
# For more, refer to the documentation: https://neptune.ai/product/model-registry
```